src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -384,9 +402,149 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+            for (uint32_t j = i + 1; j < _CurrentBellSounds.Size(); j++)
+            {
+                auto soundB = _CurrentBellSounds.GetAt(j);
+                if (soundA.DisplayPath() == soundB.DisplayPath())
+                {
+                    get_self<BellSoundViewModel>(_CurrentBellSounds.GetAt(i))->ShowDirectory(true);
+                    get_self<BellSoundViewModel>(_CurrentBellSounds.GetAt(j))->ShowDirectory(true);
+                }
+            }
+        }
+    }
+
+    BellSoundViewModel::BellSoundViewModel(hstring path, const Windows::UI::Core::CoreDispatcher& dispatcher) :
+        _Path{ path },
+        _dispatcher{ dispatcher }
+    {
+        _CheckIfFileExists();
+    }
+
+    safe_void_coroutine BellSoundViewModel::_CheckIfFileExists()
+    {
+        co_await winrt::resume_background();
+        const auto exists = std::filesystem::exists(std::wstring_view{ _Path });
+
+        co_await winrt::resume_foreground(_dispatcher);
+        FileExists(exists);
+    }
+
+    hstring BellSoundViewModel::DisplayPath() const
+    {
+        if (FileExists())
+        {
+            // filename
+            const std::filesystem::path filePath{ std::wstring_view{ _Path } };
+            return hstring{ filePath.filename().wstring() };
+        }
+        return _Path;
+    }
+
+    hstring BellSoundViewModel::SubText() const
+    {
+        if (FileExists())
+        {
+            // Directory
+            const std::filesystem::path filePath{ std::wstring_view{ _Path } };
+            return hstring{ filePath.parent_path().wstring() };
+        }
+        return RS_(L"Profile_BellSoundNotFound");
+    }
+
+    hstring ProfileViewModel::BellSoundPreview()
+    {
+        const auto& currentSound = BellSound();
+        if (!currentSound || currentSound.Size() == 0)
+        {
+            return RS_(L"Profile_BellSoundPreviewDefault");
+        }
+        else if (currentSound.Size() == 1)
+        {
+            std::filesystem::path filePath{ std::wstring_view{ currentSound.GetAt(0) } };
+            return hstring{ filePath.filename().wstring() };
+        }
+        return RS_(L"Profile_BellSoundPreviewMultiple");
+    }
+
+    void ProfileViewModel::RequestAddBellSound(hstring path)
+    {
+        // If we were inheriting our bell sound,
+        // copy it over to the current layer and apply modifications
+        _PrepareModelForBellSoundModification();
+
+        _CurrentBellSounds.Append(winrt::make<BellSoundViewModel>(path, _dispatcher));
+        _profile.BellSound().Append(path);
+        _NotifyChanges(L"CurrentBellSounds");
+    }
+
+    void ProfileViewModel::RequestDeleteBellSound(const Editor::BellSoundViewModel& vm)
+    {
+        uint32_t index;
+        if (_CurrentBellSounds.IndexOf(vm, index))
+        {
+            // If we were inheriting our bell sound,
+            // copy it over to the current layer and apply modifications
+            _PrepareModelForBellSoundModification();

```

Comment:
technically this is fine, but it does scare me - we're checking the membership in ONE array, and then editing a SECOND array (!)